Adding issue template suggestions (#2416)
inmanta/inmanta-core: .github/ISSUE_TEMPLATE/bug-report.md

---
name: Bug report
about: Unexpected behavior? Tell us more
title: ''
labels: bug
assignees: ''

---

# Expected Behavior (short explanation)

Please describe the behavior you are expecting

# Current Behavior

What is the current behavior?

# Failure Information

Please help provide information about the failure.

## Steps to Reproduce

1. step 1
2. step 2
3. you get it...

## Context

Please provide any relevant information about your setup. 

* Inmanta version:

> * Operating System:

Reviewer: Maybe add _python version_ to the list as well? What about PostgreSQL version?

Author: Added it in https://github.com/inmanta/inmanta/pull/2416/commits/4f10b989b54c0c76c2a13c81416026cef61a37ed

Author: What about PostgreSQL version @bartv ? Would that be relevant?

Reviewer 2: yes

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/109c0803e5788a67f3414672083d954ea8283964

> * Operating System:

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/4f10b989b54c0c76c2a13c81416026cef61a37ed

## Failure Logs
<details>

> <summary>Click to deploy the logs</summary>

Reviewer 3: I think you mean this:
```suggestion
<summary>Click to display the logs</summary>
```

---

> name: Bug report

Reviewer 2: I think we also need to ask for logs (server, compiler, agent) if they are relevant. Not sure I which section this belongs.

> about: Unexpected behavior? Tell us more

Reviewer 2: This is used as a caption to the "Bug report" title. I like the version of the default github template a lot more. I think the tone of this is a bit informal. The default github proposal is:

```suggestion
about: Create a report to help us improve
```

title: ''
labels: bug
assignees: ''

---

# Describe your issue

> (Tell us what the problem is in a few words)  

Reviewer 2: This is obvious from the title no? The idea of the template is to guide the user into creating a better report. Again, the default github template is a lot clearer there:

```suggestion
A clear and concise description of what the bug is.
```

> *Ex: As an end user, when I access the console and see a service inventory, if attributes have too long values, I can't see any of the right buttons anymore*

Reviewer 2: And then the example is not a very good one, because it does not use simple English sentences (I even think the sentence is incorrect). 

Same content, but a lot more concise and easier to read: "Long attributes will force the buttons off screen in the service inventory."

# Steps to reproduce

> 1. step 1

Reviewer 2: I do not really understand why examples are given in the other sections but here you just have step1, step2, ... which is obvious from the numbers of course.

I do understand that it depends on the project that it depends. However, the current examples are all based on the same issue. So why not:

1. Go to the console
2. Go to the service catalog by clicking service catalog in the menu
3. Go to the service inventory of a service
4. When a service instance exists with a very long attribute the buttons on the right are no longer visible.

2. step 2
3. ...
4. result

# Expected behavior

(Tell us what you expected this list of steps to do)  
*Ex: On the right side of the service I should see 4 buttons: show resources, edit, delete and diagnose*

# Work around (if any)

> (You can add detail about the resolution of this issue here)  

Reviewer 2: resolution is not the same as workaround, solving it is of course a workaround as well.

```suggestion
Provide any details about a workaround or solution if it exists.
```

> *Ex: Buy 8K monitor to every user so that there is room for such long test*

Reviewer 4: I wouldn't go for silly jokes in the template. Just as example say: none.

Author: https://github.com/inmanta/inmanta/pull/2416/commits/de7173bced89215bb959048a094397b161ebae34

# Your environment

> Please provide the following information:

Reviewer 2: We need to customize this. For this repo this is ok. For console and dashboard it should contain at least the browser version and python and pg are not useful.

For modules it should contain questions about the systems it interfaces with.

In that light I do not understand the example used, because that would go into the web console repo and is not relevant for this repo.

 - Current Inmanta version: 
 - Operating System: 
 - Python version:
 - PostgreSQL version:

> Using Linux? You can replace the list by the output of the following command:  

Reviewer 2: Is this useful? Because if you are using a virtualenv and have installed pg10 from redhat software collections half of the commands will not work.

> ---

Reviewer 4: What does this header here do?

Author: It specifies the name of the template, a small description the user will see when having to pick one, and other things like default title, automatically attached labels, etc.

name: Bug report
about: Unexpected behavior? Tell us more
title: ''
labels: bug
assignees: ''

---

> # Describe your issue

Reviewer 5: It seems like the description and the  `{ 'steps to reproduce', 'expected behavior', 'actual behavior' }` set have a lot in common.
I'm not sure of the best solution, maybe we can just let the issue authors decide what they want to emphasize more, and leave them as they are

Author: In https://github.com/inmanta/inmanta/pull/2416/commits/97df07c2c9d53836b23046f865629bacef6439c7 I removed the actual behavior and suggested that this would come at the end of steps to reproduce, is this better?

> (Tell us what the problem is in a few word)

Reviewer 5: ```suggestion
(Tell us what the problem is in a few words)
```

Author: Fixed it in https://github.com/inmanta/inmanta/pull/2416/commits/47d514a34f85f514a40a54849bef9082cb18ade9

> # Your environment

Reviewer 4: This I would move to the bottom. 

It is in fact the least important part, so it should not take up the top

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/97df07c2c9d53836b23046f865629bacef6439c7

> Please provide the following informations:

Reviewer 5: ```suggestion
Please provide the following information:
```

Author: Fixed it in https://github.com/inmanta/inmanta/pull/2416/commits/47d514a34f85f514a40a54849bef9082cb18ade9

 - Current Inmanta version: 
 - Operating System: 
 - Python version:
 - PostgreSQL version:

Using Linux? You can replace the list by the output of the following command:  
echo " - $(inmanta --version)"; \
echo " - OS: $(source /etc/os-release; echo $PRETTY_NAME)"; \
echo " - Python version: $(python3 --version)"; \
echo " - PostreSQL version: $(psql --version)"

# Steps to reproduce

1. step 1
2. step 2

> 3. you get it...

Reviewer 4: ```suggestion
```

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/97df07c2c9d53836b23046f865629bacef6439c7

# Expected behavior

(Tell us what you expected this list of steps to do)

# Actual behavior

> (Tell us what actually happened)

Reviewer 4: A bit redundant wrt to Steps to reproduce I think?

Author: In https://github.com/inmanta/inmanta/pull/2416/commits/97df07c2c9d53836b23046f865629bacef6439c7 I remove this section and add one more step in *steps to reproduce* to suggest that it would go there, is this alright?

> # (Optional) You already found the solution to this problem?

Reviewer 4: ```suggestion
# Work around (if any)
```

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/97df07c2c9d53836b23046f865629bacef6439c7

(You can add detail about the resolution of this issue here)

> # (Optional) Are you willing to contribute and try solving this issue?

Reviewer 4: I would remove this section.

Author: Did it in https://github.com/inmanta/inmanta/pull/2416/commits/97df07c2c9d53836b23046f865629bacef6439c7

> # (Optional) Are you willing to contribute and try solving this issue?

Reviewer 5: Even while stating it's Optional, it feels like there is only one choice (to say yes).
But that might be only my impression

Author: I removed it in https://github.com/inmanta/inmanta/pull/2416/commits/97df07c2c9d53836b23046f865629bacef6439c7

---
name: Bug report
about: Unexpected behavior? Tell us more
title: ''
labels: bug
assignees: ''

---

**Self checklist**

 - [ ] This issue hasn't been already created
 - [ ] Everything is sound and clear, all sections are filled-up
 - [ ] Relevant labels have been added

# Expected Behavior (short explanation)

Please describe the behavior you are expecting

# Current Behavior

What is the current behavior?

> # Failure Information

Reviewer 4: How is current behavior different from Failure Information
?

Author: Failure information is just a section that contains the *steps to reproduce*, the *context* and the *failure logs,* whereas the *current behavior* is asking for explanation about what is going on.  Perhaps this is not clear

Author: Is it more clear now? https://github.com/inmanta/inmanta/pull/2416/commits/5090b071922f322cee876f9d6288b7279514d3b2

Please help provide information about the failure.

## Steps to Reproduce

1. step 1
2. step 2
3. you get it...

## Context

Please provide any relevant information about your setup. 

* Inmanta version:
* Operating System:
* Python version:

> ## Failure Logs

Reviewer 4: how is failure logs different from Failure Information and context

Author: *Failure logs* should contain any relevant trace (in case of error), debug execution steps (if relevant) or any output from the software (here `inmanta`).

The *context* might contain global information (like if the error occured since the last system update).

Both *failure logs* and *context* are a part of *failure information*

Author: Is it more clear now? https://github.com/inmanta/inmanta/pull/2416/commits/5090b071922f322cee876f9d6288b7279514d3b2

Reviewer 4: Why is this not part of the Current Behavior or Steps to Reproduce (both seem natural places to put a stacktrace?)

Author: Structure has been updated here: https://github.com/inmanta/inmanta/pull/2416/commits/109c0803e5788a67f3414672083d954ea8283964

---
name: Bug report
about: Unexpected behavior? Tell us more
title: ''
labels: bug
assignees: ''

---

**Self checklist**

 - [ ] I checked that this issue hasn't been created before

>  - [ ] I added any relevant label to the issue

Reviewer 4: How should a person, external to inmanta know what the relevant label is?

Author: Checklist has been removed in https://github.com/inmanta/inmanta/pull/2416/commits/35fb7151fd7c0a968dadf87696f42179d272e650

# Expected Behavior (short explanation)

Please describe the behavior you are expecting

# Current Behavior

> Please describe the corrent behavior

Reviewer: ```suggestion
Please describe the current behavior
```
